strongbox-storage/strongbox-storage-core/src/main/java/org/carlspring/strongbox/storage/repository/Repository.java

```diff
@@ -99,13 +104,13 @@
     @XmlElementRef
     private CustomRepositoryConfiguration repositoryConfiguration;
 
-    @XmlElement(name = "repository")
-    @XmlElementWrapper(name = "group")
-    private Set<String> groupRepositories = new LinkedHashSet<>();
+    @XmlElement(name = "group")
+    @XmlJavaTypeAdapter(RepositoryGroupsAdapter.class)
+    private Map<String, String> groupRepositories = new LinkedHashMap<>();
```

Comment:
@fuss86 : 

Could you please add some documentation in a follow-up pull as to why this is a map?